Code/GraphMol/SynthonSpaceSearch/SynthonSpaceFingerprintSearcher.cpp

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright (C) David Cosgrove 2024.
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <DataStructs/BitOps.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.h>
+#include <GraphMol/SynthonSpaceSearch/SynthonSpaceFingerprintSearcher.h>
+
+namespace RDKit::SynthonSpaceSearch {
+
+SynthonSpaceFingerprintSearcher::SynthonSpaceFingerprintSearcher(
+    const ROMol &query, const FingerprintGenerator<std::uint64_t> &fpGen,
+    const SynthonSpaceSearchParams &params, SynthonSpace &space)
+    : SynthonSpaceSearcher(query, params, space), d_fpGen(fpGen) {
+  if (!getSpace().hasFingerprints() ||
+      getSpace().getSynthonFingerprintType() != fpGen.infoString()) {
+    getSpace().buildSynthonFingerprints(fpGen);
```

Comment:
I think it's worth showing a warning message at this point... this operation may take a while